core/src/main/scala/io/prediction/core/CleanedDataSource.scala

```diff
@@ -103,12 +106,29 @@ trait CleanedDataSource {
       } ++ events.filter(!isSetEvent(_))
   }
 
+  //TODO ensure most recent is preserved
   def removePDuplicates(sc: SparkContext, rdd: RDD[Event]): RDD[Event] = {
```

Comment:
this is pretty important given that the event will be gone prematurely if we don't de-dup to the most recent. Any idea if this is a real problem?